openedx/adg/lms/applications/models.py

```diff
@@ -0,0 +1,118 @@
+"""
+All models for applications app
+"""
+from django.contrib.auth.models import User
+from django.core.validators import FileExtensionValidator
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from model_utils.models import TimeStampedModel
+
+from openedx.adg.common.util.date_utils import month_choices, year_choices
+
+
+class ApplicationHub(TimeStampedModel):
+    """
+    Model for status of all required parts of user application submission.
+    """
+    user = models.OneToOneField(User, related_name='application_hub', on_delete=models.CASCADE, )
+    is_prerequisite_courses_passed = models.BooleanField(default=False, )
+    is_application_submitted = models.BooleanField(default=False, )
+    is_assessment_completed = models.BooleanField(default=False, )
+
+    class Meta:
+        app_label = 'applications'
+
+    def __str__(self):
+        return 'User {user_id}, application status id={id}'.format(user_id=self.user.id, id=self.id)
+
+
+class UserApplication(TimeStampedModel):
+    """
+    Model for status of all required parts of user application submission.
+    """
+    user = models.OneToOneField(User, related_name='application', on_delete=models.CASCADE, )
+    city = models.CharField(verbose_name=_('City'), max_length=255, )
+    organization = models.CharField(verbose_name=_('Organization'), max_length=255, blank=True, )
+    linkedin_url = models.URLField(verbose_name=_('LinkedIn URL'), max_length=255, blank=True, )
+    resume = models.FileField(
+        max_length=500, blank=True, null=True, upload_to='files/resume/',
+        validators=[FileExtensionValidator(['pdf', 'doc', 'jpg', 'png'])],
+        help_text=_('Accepted extensions: .pdf, .doc, .jpg, .png'),
+    )
+    cover_letter_file = models.FileField(
+        max_length=500, blank=True, null=True, upload_to='files/cover_letter/',
+        validators=[FileExtensionValidator(['pdf', 'doc', 'jpg', 'png'])],
+        help_text=_('Accepted extensions: .pdf, .doc, .jpg, .png'),
+    )
+    cover_letter = models.TextField(blank=True, )
+
+    class Meta:
+        app_label = 'applications'
+        verbose_name = _('User Application')
+        verbose_name_plural = _('User Applications')
+
+    def __str__(self):
+        return 'UserApplication {id}, for user {email}'.format(id=self.id, email=self.user.email)
+
+
+class UserApplicationCommon(TimeStampedModel):
+    month_choices = month_choices(default_title='Month')
+    year_choices = year_choices(default_title='Year')
+
+    user_application = models.ForeignKey(
+        'UserApplication', related_name='%(app_label)s_%(class)ss', related_query_name='%(app_label)s_%(class)s',
+        on_delete=models.CASCADE,
+    )
+    date_started_month = models.CharField(choices=month_choices, max_length=2, )
```

Comment:
We should either save integers both incase of month and year or string. I guess I requested something like this last day but not sure.